core/Segment.php

```diff
@@ -110,6 +110,35 @@ public function __construct($segmentCondition, $idSites)
         }
     }
 
+    private function getAvailableSegments()
+    {
+        // segment metadata
+        if (empty($this->availableSegments)) {
+            $this->availableSegments = API::getInstance()->getSegmentsMetadata($this->idSites, $_hideImplementationData = false);
```

Comment:
This shouldn't change between different `Segment` instances correct? Perhaps this logic can be moved to an object in DI so `API.getSegmentsMetadata` is not called more than once? Actually, maybe `API.getSegmentsMetadata` could store the result in a cache?